hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -234,15 +225,25 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     String fromName = from.name();
 
     try {
-      Table table = clients.run(client -> client.getTable(fromDatabase, fromName));
-      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, from));
-
-      table.setDbName(toDatabase);
-      table.setTableName(to.name());
+      Table fromTable = clients.run(client -> client.getTable(fromDatabase, fromName));
+      HiveOperationsBase.validateTableIsIceberg(fromTable, fullTableName(name, from));
+      Optional<Table> toTable =
+          HiveCatalogUtil.loadHmsTableIfPresent(clients, toDatabase, to.name());
+      if (toTable.isPresent()) {
+        if (toTable.get().getTableType().equalsIgnoreCase(TableType.VIRTUAL_VIEW.name())) {
```

Comment:
Question: Would this be this be easier to read?
```
    if (toTable.isPresent()) {
      TableType type = toTable.get().getTableType();
      throw new org.apache.iceberg.exceptions.AlreadyExistsException(
          "Cannot rename %s to %s. %s already exists",
          from,
          to,
          type.equalsIgnoreCase(TableType.VIRTUAL_VIEW.name()) ? "View" : "Table");
    }
```

Nothing you must change, but seems a bit more natural to me.